server/task_check_executor_lgtm.go

```diff
@@ -0,0 +1,143 @@
+				Namespace: api.BBNamespace,
+				Code:      common.Internal.Int(),
+				Title:     fmt.Sprintf("Failed to find task %v", taskCheckRun.TaskID),
+				Content:   err.Error(),
+			},
+		}, nil
+	}
+
+	issue, err := server.store.GetIssueByPipelineID(ctx, task.PipelineID)
+	if err != nil {
+		return []api.TaskCheckResult{}, common.Wrap(err, common.Internal)
+	}
+	if issue == nil {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusError,
+				Namespace: api.BBNamespace,
+				Code:      common.Internal.Int(),
+				Title:     fmt.Sprintf("Failed to find issue by pipelineID %v", task.PipelineID),
+				Content:   err.Error(),
+			},
+		}, nil
+	}
+
+	if issue.Project.LGTMCheckSetting.Value == api.LGTMValueDisabled {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusSuccess,
+				Namespace: api.BBNamespace,
+				Code:      common.Ok.Int(),
+				Title:     "Skip check",
+				Content:   "This check is disabled.",
+			},
+		}, nil
+	}
+
+	activityType := string(api.ActivityIssueCommentCreate)
+	activityList, err := server.store.FindActivity(ctx, &api.ActivityFind{
+		TypePrefix:  &activityType,
+		ContainerID: &issue.ID,
+	})
+	if err != nil {
+		return []api.TaskCheckResult{}, common.Wrap(err, common.Internal)
+	}
+
+	ok, err := checkLGTMcomments(ctx, server.store, activityList, issue)
+	if err != nil {
+		return []api.TaskCheckResult{}, errors.Wrap(err, "failed to check LGTM comments")
+	}
+	if ok {
+		return []api.TaskCheckResult{
+			{
+				Status:    api.TaskCheckStatusSuccess,
+				Namespace: api.BBNamespace,
+				Code:      common.Ok.Int(),
+				Title:     "OK",
+				Content:   "Valid LGTM found",
+			},
+		}, nil
+	}
+
+	return []api.TaskCheckResult{
+		{
+			Status:    api.TaskCheckStatusError,
+			Namespace: api.BBNamespace,
+			Code:      common.NotFound.Int(),
+			Title:     "Not Found",
+			Content:   "Valid LGTM NOT found",
+		},
+	}, nil
+}
+
+func checkLGTMcomments(ctx context.Context, store *store.Store, activityList []*api.Activity, issue *api.Issue) (bool, error) {
+	for _, activity := range activityList {
+		ok, err := isCommentLGTM(ctx, store, activity, issue)
+		if err != nil {
+			return false, err
+		}
+		if ok {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func isCommentLGTM(ctx context.Context, store *store.Store, activity *api.Activity, issue *api.Issue) (bool, error) {
+	if activity.Comment != "LGTM" {
+		return false, nil
+	}
+	// LGTM by oneself doesn't count.
+	if activity.CreatorID == issue.CreatorID {
+		return false, nil
+	}
+	member, err := store.GetProjectMember(ctx, &api.ProjectMemberFind{
+		PrincipalID: &activity.CreatorID,
+		ProjectID:   &issue.ProjectID,
+	})
+	if err != nil {
+		return false, err
+	}
+	if member == nil {
+		return false, nil
+	}
+	switch issue.Project.LGTMCheckSetting.Value {
+	case api.LGTMValueProjectMember:
+		return true, nil
+	case api.LGTMValueProjectOwner:
+		return member.Role == string(api.Owner), nil
```

Comment:
Move line 126 here. It's the only place using `member`.

Reply:
The query is needed because we want to check if the commenter is a project member.